compiler/next/include/chpl/uast/DoWhile.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_UAST_DOWHILE_H
+#define CHPL_UAST_DOWHILE_H
+
+#include "chpl/queries/Location.h"
+#include "chpl/uast/Loop.h"
+
+namespace chpl {
+namespace uast {
+
+
+/**
+  This class represents a do-while loop. For example:
+
+  \rst
+  .. code-block:: chapel
+
+      // Example 1:
+      var i = 0;
+      do {
+        writeln(i);
+        i += 1;
+      } while i < 5;
+
+  \endrst
+
+ */
+class DoWhile final : public Loop {
+ private:
+  DoWhile(ASTList children, int8_t conditionChildNum)
+    : Loop(asttags::DoWhile, std::move(children),
+           (conditionChildNum + 1),
+           /*usesDo*/ false),
```

Comment:
See my comment on the parser rule - we do need `usesDo` to be passed through from the parser.